libraries/AP_NavEKF2/AP_NavEKF2_core.cpp

```diff
@@ -551,6 +552,28 @@ void NavEKF2_core::UpdateFilter(bool predict)
 #if EK2_DISABLE_INTERRUPTS
     irqrestore(istate);
 #endif
+
+    /*
+      this is a check to cope with a vehicle sitting idle on the
+      ground and getting over-confident of the state. The symptoms
+      would be "gyros still settling" when the user tries to arm. In
+      that state the EKF can't recover, so we do a hard reset and let
+      it try again.
+     */
+    if (filterStatus.value != 0) {
```

Comment:
if EKF is always unhealthy after power up, it can't be reset.

Reply:
yes, that is deliberate. This is specifically meant for the case that the EKF goes unhealthy after initially being healthy